From ea0f3a9b7f9887e04c2de8e7801ddd977aaf21bb Mon Sep 17 00:00:00 2001
From: Drew Moseley <drew_moseley@mentor.com>
Date: Sat, 5 Jul 2014 18:53:38 -0400
Subject: [PATCH 04/13] init-install-efi.sh: Copy existing files from USB /boot
 into boot partition.

Copy any files that exist in the USB rootfs "/boot" directory into the newly
created boot partition.  This requires increasing the size to 250MB.

Signed-off-by: Drew Moseley <drew_moseley@mentor.com>
---
 meta/recipes-core/initrdscripts/files/init-install-efi.sh | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

Upstream-Status: Inappropriate [configuration]

diff --git a/meta/recipes-core/initrdscripts/files/init-install-efi.sh b/meta/recipes-core/initrdscripts/files/init-install-efi.sh
index e65deea..44965bd 100644
--- init-install-efi.sh
+++ init-install-efi.sh
@@ -8,8 +8,8 @@
 
 PATH=/sbin:/bin:/usr/sbin:/usr/bin
 
-# We need 20 Mb for the boot partition
-boot_size=20
+# We need 250 Mb for the boot partition
+boot_size=250
 
 # 5% for swap
 swap_ratio=5
@@ -164,11 +164,14 @@ if [ -d /tgt_root/etc/ ] ; then
 	echo "/dev/${device}" >> /tgt_root/etc/udev/mount.blacklist
     fi
 fi
-
-umount /src_root
  
 # Handling of the target boot partition
 mount $bootfs /boot
+
+# Copy existing /boot files into boot partition
+cp -aH /src_root/boot/* /boot
+umount /src_root
+
 echo "Preparing boot partition..."
 
 EFIDIR="/boot/EFI/BOOT"
-- 
1.9.1

